tests/e2e/tests/simple/simple1_test.go

```diff
@@ -128,33 +114,30 @@ func TestSvc2Svc(t *testing.T) {
 	if len(podList) != 2 {
 		t.Fatalf("Unexpected to get %d pods when expecting 2. got %v", len(podList), podList)
 	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-	g, ctx := errgroup.WithContext(ctx)
-
-	log.Infof("Configuration readiness pre-check from %v to http://echosrv.%s:8080/echo", podList, ns)
-	for i := range podList {
-		pod := podList[i]
-		g.Go(func() error {
-			for {
-				select {
-				case <-ctx.Done():
-					return ctx.Err()
-				default:
-					_, err := util.ShellContext(ctx, "kubectl exec -n %s %s -c echosrv -- /usr/local/bin/fortio curl http://echosrv.%s:8080/echo", ns, pod, ns) // nolint: lll
-					if err == nil {
-						return nil
-					}
-					time.Sleep(time.Second)
-				}
+	// Check that both pods are ready before doing a (small) load test
+	log.Infof("Configuration readiness pre-check from %v to http://echosrv:8080/echo", podList)
+	timeout := time.Now().Add(timeToWaitForPods)
```

Comment:
any reason to avoid `util.CheckPodsRunning(ns)` ?

Reply:
No good reason: me not knowing about it - but does that check that envoy is ready though ?

Reply:
that one seems to wait 30s at least which is rather slow and only checks pods, but I'll add it in front and change its timing

Reply:
it doesn't. it is almost like we could use some liveliness and readiness probes ;).